### ISSUE TITLE
blocking-open-call-in-async-function (ASYNC230) does not trigger on builtins.open

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ruff#28012]
```python
import builtins

async def f():
    open("x")           # ASYNC230
    builtins.open("x")  # Missed
    input()             # ASYNC250
    builtins.input()    # ASYNC250
```

These two rules are closely analogous but only one catches use through `builtins`.

Incidentally this also affects the `unspecified-encoding` error.
